backend/prisma/schema.prisma

```diff
@@ -15,20 +15,21 @@ datasource db {
 }
 
 model User {
-  id                Int                   @id @default(cuid())
+  id                String                @id @default(cuid())
   email             String                @unique
   password          String
   
-  profile           UserProfile?          @relation(fields: [id], references: [userId])
-  sensitiveData     UserSensitiveData?    @relation(fields: [id], references: [userId])
-  financialRecords  FinancialData[]       @relation("UserFinancialRecords")      
+  profile           UserProfile?          @relation("UserProfile")
+  sensitiveData     UserSensitiveData?    @relation("UserSensitiveData")
+  financialRecords  FinancialData[]       @relation("FinancialData")      
   transactions      Transaction[]         @relation("UserTransactions")
+  notifications     Notification[]        @relation("Notification")
 }
 
 model UserProfile {
-  id        String    @id @default(cuid(id))
+  id        String    @id @default(cuid())
   userId    String    @unique
-  user      User      @relation(fields: [userId, references: [id]])
+  user      User      @relation("UserProfile", fields: [userId], references: [id])
 
   ageRange  String?
   gender    String?
```

Comment:
it's good practice to separate data model for security and privacy in the real world, but I'm worried if might be overkill for this project. 🤔